src/copilot-cli/skills/golden-principles/scripts/scan_principles.py

```
@@ -107,6 +108,38 @@ def get_repo_files(directory: str) -> list[str]:
+
+    Raises:
+        ValueError: ``base`` is empty or starts with ``-`` (CWE-88 argument
+            injection: a leading dash would be parsed by git as an option).
+        RuntimeError: git is unavailable or the diff command fails (for example
+            an unknown base). A git failure must not be mistaken for an empty
+            diff, which would let a standards pre-flight pass without scanning.
+    """
+    if not base or base.startswith("-"):
+        raise ValueError(f"invalid --diff-scope base: {base!r}")
+    try:
+        result = subprocess.run(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)